### PR TITLE
For #19739 - Adds start tabsTray with multiSelect support

### PR DESCRIPTION
**Pull request by codrut-topliceanu**
For https://github.com/mozilla-mobile/fenix/issues/19739
### Pull Request checklist

- [x] **Tests**: This PR includes no tests as it is only a minor change
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made.
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md).
### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

https://user-images.githubusercontent.com/60002907/121353308-61bb1680-c936-11eb-8fac-bdfb3f203d68.mp4

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFragment.kt [hunk 1]
```diff
@@ -278,6 +278,10 @@ class TabsTrayFragment : AppCompatDialogFragment() {
             owner = this,
             view = view
         )
+
+        if (requireArguments().getBoolean(EXTRA_MULTI_SELECT)) {
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFragment.kt:282]: Does this check need to be done here or can it go into the `initialState` when creating the `TabsTrayStore` on line 96?

Comment (codrut-topliceanu) [reply app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFragment.kt:282]: Hmm... I'll run a few tests and see. I added it last as a precaution.

Comment (jonalmeida) [reply app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFragment.kt:282]: What's nice about using Kotlin flow's and `ifChanged` is that we get the initial state _and_ the changed results from our data source.

This means we don't have to worry about timing issues (assuming our features are not dependant on each other).

Comment (codrut-topliceanu) [reply app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFragment.kt:282]: Sorry for the delay, got caught up yesterday with another PR. 
Yeah, I agree. Will update this asap.

**Review by jonalmeida — APPROVED** [review]

**Pull request merged**